modules/sonar/Makefile

```diff
@@ -0,0 +1,37 @@
+## #####################################
+##
+## Module: Sonar
+##
+## This makefile contains common instructions to scan projects with SonarQube
+
+## Variables
+
+## SONAR_AUTH_TOKEN is the token you generate to authenticate with your Sonar 
+## deployment. You can see how to generate and use one here: 
+## https://docs.sonarqube.org/latest/user-guide/user-token/
+## THIS TOKEN SHOULD BE KEPT A SECRET
+
+## SONAR_URL is the URL for your Sonar deployment. While showing this value is 
+## not necessarily dangerous, it is recommended that you keep it a secret
+
+##
+## Sonar Module Targets
+
+## Scans project and sends results to your Sonar deployment
+.PHONY: sonar
+sonar:
+ifndef GO_MODULE
```

Comment:
Sorry I mislead you in the previous review of this check. The correct way is to use the `require-VAR` target as dependency. This is defined in the `meta` module.

The check would become `require-SONAR_AUTH_TOKEN`. By doing that you are also free from the if statement and can just run the docker command.

Also, please always escape variables with quotes. Without quotes scripts tend to break with directories containing spaces.

I also noticed that the GO_MODULE is used just to mount a directory inside the docker container. Can we get rid of it and just mount the code in a fixed directory?

I was thinking in something like this for the target:

```
## Scans project and sends results to your Sonar deployment
.PHONY: sonar-scan
sonar-scan: | require-SONAR_AUTH_TOKEN require-SONAR_URL require-GO_PROJECT
	docker run \
		--rm \
		--env SONAR_HOST_URL="${SONAR_URL}" \
		--env SONAR_LOGIN="${SONAR_AUTH_TOKEN}" \
		--env SONAR_SCANNER_OPTS="-Dsonar.projectKey=${GO_PROJECT} -Dsonar.sources=. -Dsonar.exclusions=**/*_test.go,**/*.xml,**/*.xsd,**/*.html -Dsonar.tests=. -Dsonar.test.inclusions=**/*_test.go -Dsonar.go.coverage.reportPaths=.cover/cover.out -Dsonar.projectVersion=${VERSION}" \
		--volume "$(CURDIR):/src" \
		--workdir /src \
		sonarsource/sonar-scanner-cli

```
Would this work?

PS: I also changes the target name to sonar-scan because all other modules use the name of the module as a prefix for the targets.

Reply:
oh, yes, this looks better, i tested and it worked like a charm